plugin/src/leiningen/cljs_lambda/aws.clj

```diff
@@ -26,31 +25,45 @@
     args/*region*      (assoc :region  (name args/*region*))
     args/*aws-profile* (assoc :profile (name args/*aws-profile*))))
 
-(defn update-dead-letter-value [{:keys [dead-letter] :as fn-spec}]
-  (if-not dead-letter
-    fn-spec
-    (merge fn-spec {:dead-letter (str "TargetArn=" dead-letter)})))
-
-(defn update-vpc-value [{:keys [vpc] :as fn-spec}]
-  (if-not vpc
-    fn-spec
-    (let [subnets (:subnets vpc)
-          security-groups (:security-groups vpc)]
-      (merge fn-spec {:vpc (string/join
-                             ["SubnetIds=["
-                              (string/join "," subnets)
-                              "],SecurityGroupIds=["
-                              (string/join "," security-groups)
-                              "]"])}))))
+(defmulti ->cli-arg-value
+  (fn [k v] k))
+
+(defmethod ->cli-arg-value :vpc-config [k v]
+  (let [subnets (:subnets v)
+        security-groups (:security-groups v)]
+    (string/join
+      ["SubnetIds=["
+       (string/join "," subnets)
+       "],SecurityGroupIds=["
+       (string/join "," security-groups)
+       "]"])))
+
+(defmethod ->cli-arg-value :environment [k v]
+  (str
+    "Variables={"
+    (string/join
+      ","
+      (for [[k v] v]
+        (str k "=" v)))
```

Comment:
@zrzka Thanks, this looks good to me - two tiny requests:

- Could we do `(name k)` here, so that keywords can be supplied?
- The second call to `str` in the `dead-letter-config` method below is superfluous.